com.unity.render-pipelines.core/Editor/Volume/VolumeComponentEditor.cs

```diff
@@ -626,13 +609,149 @@ protected void DrawOverrideCheckbox(SerializedDataParameter property)
+
+                Init(property, label, editor);
+            }
+
+            /// <summary>
+            /// Constructor
+            /// </summary>
+            /// <param name="property">The property that will be drawn</param>
+            /// <param name="label">The label of this property</param>
+            /// <param name="editor">The editor that will draw it</param>
+            public OverridablePropertyScope(SerializedDataParameter property, string label, VolumeComponentEditor editor)
+            {
+                disabledScope = null;
+                haveCustomOverrideCheckbox = false;
+                drawer = null;
+                displayed = false;
+                isAdditionalProperty = false;
+                this.label = EditorGUIUtility.TrTextContent(label);
+                this.editor = editor;
+
+                Init(property, this.label, editor);
+            }
+
+            void Init(SerializedDataParameter property, GUIContent label, VolumeComponentEditor editor)
+            {
+                // Below, 3 is horizontal spacing and there is one between label and field and another between override checkbox and label
+                EditorGUIUtility.labelWidth -= Styles.overrideCheckboxWidth + Styles.overrideCheckboxOffset + 3 + 3;
+
+                isAdditionalProperty = property.GetAttribute<AdditionalPropertyAttribute>() != null;
+                displayed = !isAdditionalProperty || editor.BeginAdditionalPropertiesScope();
+
+                s_ParameterDrawers.TryGetValue(property.referenceType, out VolumeParameterDrawer vpd);
+                drawer = vpd;
+
+                //never draw override for embedded class/struct
+                haveCustomOverrideCheckbox = (displayed && !(drawer?.IsAutoProperty() ?? true))
+                    || VolumeParameter.IsObjectParameter(property.referenceType);
+
+                if (displayed)
+                    editor.HandleDecorators(property, label);
+
+                if (!haveCustomOverrideCheckbox && displayed)
+                {
+                    EditorGUILayout.BeginHorizontal();
+                    editor.DrawOverrideCheckbox(property);
+
+                    disabledScope = new EditorGUI.DisabledScope(!property.overrideState.boolValue);
+                }
+            }
+
+            void IDisposable.Dispose()
+            {
+                disabledScope?.Dispose();
+
+                if (!haveCustomOverrideCheckbox && displayed)
+                    EditorGUILayout.EndHorizontal();
+
+                if (isAdditionalProperty)
+                    editor.EndAdditionalPropertiesScope();
+
+                EditorGUIUtility.labelWidth += Styles.overrideCheckboxWidth + Styles.overrideCheckboxOffset + 3 + 3;
+            }
+        }
+
+        /// <summary>
+        /// Like EditorGUI.IndentLevelScope but this one will also indent the override checkboxes.
+        /// </summary>
+        protected class IndentLevelScope : GUI.Scope
```

Comment:
For me is not clear that this scope is indenting the overrides too, can we find a better name?